src/db/mysql/Schema.php

```diff
@@ -204,7 +204,7 @@ public function getDefaultBackupCommand(?array $ignoreTables = null): string
             ->addArg('--dump-date')
             ->addArg('--no-autocommit')
             ->addArg('--routines')
-            ->addArg('--default-character-set=', Craft::$app->getConfig()->getDb()->charset)
```

Comment:
Seems like ideally we'd want to make `\craft\config\DbConfig::$charset` protected and add `setCharset` so this code would work either way?

The only reason why the current method is `\craft\config\DbConfig::charset` and not `\craft\config\DbConfig::setCharset` is because of fluent config conventions. So, if we add it, one can just reference the other.

Reply:
Ok, but what about the environment overrides? Currently, they rely [on the properties being public](https://github.com/craftcms/cms/blob/5.x/src/helpers/App.php#L156). Or am I missing a trick here?